spec/controllers/stripe/stripe_controller_spec.rb

```diff
@@ -12,15 +12,15 @@
       end
 
       it 'returns on bad signature' do
-        raw_post :receive, {}, {}.to_json
+        post :receive, body: {}.to_json
```

Comment:
You might not even need to do a `to_json` ... Also, doesn't this require `params: {}` ?

Reply:
> You might not even need to do a `to_json` ... Also, doesn't this require `params: {}` ?

You do have to do `.to_json` here because body in a controller test expects a raw string; it doesn't do any parsing on its own.

No, `params:` isn't required. Internally, `body:` gets parsed as json and then into `params` at run time, just like when a body gets passed to controller action on a request.

Reply:
Ooooohhhh, it's a controller spec. Ok ok... So, I suppose we'll eventually want to migrate these to request and integrations specs as appropriate.

Reply:
> Ooooohhhh, it's a controller spec. Ok ok... So, I suppose we'll eventually want to migrate these to request and integrations specs as appropriate.

I personally think controller specs are kind of underrated. They have some advantages (but also disadvantages)